tests/main/snapshot-users/task.yaml

```diff
@@ -0,0 +1,125 @@
+summary: Check that the basic snapshots functionality works for different users
+
+prepare: |
+    snap install test-snapd-tools
+
+execute: |
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB/dirs.sh"
+
+    # use the snaps, so they create the dirs
+    test-snapd-tools.echo
+    su -p -c "$SNAP_MOUNT_DIR/bin/test-snapd-tools.echo" test
+
+    # drop in canaries for both users
+    echo "hello versioned test-snapd-tools"  > /root/snap/test-snapd-tools/current/canary.txt
+    echo "hello common test-snapd-tools" > /root/snap/test-snapd-tools/common/canary.txt
+    echo "hello versioned test-snapd-tools"  > /home/test/snap/test-snapd-tools/current/canary.txt
+    echo "hello common test-snapd-tools" > /home/test/snap/test-snapd-tools/common/canary.txt
+
+    # create different snapshots for each user
+    SET_ID_ROOT=$( snap save --users=root test-snapd-tools | cut -d\  -f1 | tail -n1 )
+    SET_ID_TEST=$( snap save --users=test test-snapd-tools | cut -d\  -f1 | tail -n1 )
+
+    # Update content on canary files
+    echo "content updated"  > /root/snap/test-snapd-tools/current/canary.txt
+    echo "content updated" > /root/snap/test-snapd-tools/common/canary.txt
+    echo "content updated"  > /home/test/snap/test-snapd-tools/current/canary.txt
+    echo "content updated" > /home/test/snap/test-snapd-tools/common/canary.txt
+
+    # create snapshots for both users
+    SET_ID_BOTH=$( snap save --users=root,test test-snapd-tools | cut -d\  -f1 | tail -n1 )
+
+    # Add more files and dirs for the test user and save snapshot for both users
+    mkdir -p /home/test/snap/test-snapd-tools/current/canary/
+    echo "content updated"  > /home/test/snap/test-snapd-tools/current/canary/canary.txt
+    SET_ID_NONE=$( snap save test-snapd-tools | cut -d\  -f1 | tail -n1 )
+
+    # check all the snapshots include the correct snap
+    snap saved --id="$SET_ID_ROOT" | MATCH test-snapd-tools
+    snap saved --id="$SET_ID_TEST" | MATCH test-snapd-tools
+    snap saved --id="$SET_ID_BOTH" | MATCH test-snapd-tools
+    snap saved --id="$SET_ID_NONE" | MATCH test-snapd-tools
+
+    # check the snapshots
+    snap check-snapshot "$SET_ID_ROOT"
+    snap check-snapshot "$SET_ID_TEST"
+    snap check-snapshot "$SET_ID_BOTH"
+    snap check-snapshot "$SET_ID_NONE"
+
+    # remove the canaries for both users
+    rm -f /root/snap/test-snapd-tools/{current,common}/canary.txt
+    rm -f /home/test/snap/test-snapd-tools/{current,common}/canary.txt
+
+    # restore the snapshot for the root user and check the files
+    snap restore "$SET_ID_ROOT" test-snapd-tools
+    test -e /root/snap/test-snapd-tools/current/canary.txt
```

Comment:
Should we check the content of the files here (and in the subsequent restores) as well?

Reply:
done